Cannot install coq.14+rc1 in local switches inside directories with dune-project

#### Description of the problem

Trying to install  `8.14+rc1` inside a opam switch in a directory with a `dune-project` file leads to an error:

```
$ opam --version       
2.1.0
$ ls                                                        
dune-project
$ cat dune-project                                                       
(lang dune 2.7)
(name test)
(using coq 0.2)
(using menhir 2.1)
$ opam switch create . ocaml-base-compiler.4.07.1
...
$ opam repo add coq-core-dev https://coq.inria.fr/opam/core-dev
...
$ opam pin coq 8.14+rc1                                        
coq is now pinned to version 8.14+rc1

The following actions will be performed:
  ∗ install ocamlfind                1.9.1     [required by coq]
  ∗ install conf-gmp                 3         [required by zarith]
  ∗ install conf-findutils           1         [required by coq]
  ∗ install ocaml-secondary-compiler 4.08.1-1  [required by ocamlfind-secondary]
  ∗ install zarith                   1.12      [required by coq]
  ∗ install ocamlfind-secondary      1.9.1     [required by dune]
  ∗ install dune                     2.9.1     [required by coq]
  ∗ install coq                      8.14+rc1*
===== ∗ 8 =====
Do you want to continue? [Y/n] y

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
∗ installed conf-findutils.1
∗ installed conf-gmp.3
⬇ retrieved ocaml-secondary-compiler.4.08.1-1  (cached)
⬇ retrieved coq.8.14+rc1  (cached)
⬇ retrieved dune.2.9.1  (cached)
⬇ retrieved ocamlfind.1.9.1  (cached)
⬇ retrieved ocamlfind-secondary.1.9.1  (cached)
⬇ retrieved zarith.1.12  (cached)
∗ installed ocamlfind.1.9.1
∗ installed zarith.1.12
∗ installed ocaml-secondary-compiler.4.08.1-1
∗ installed ocamlfind-secondary.1.9.1
∗ installed dune.2.9.1
[ERROR] The compilation of coq.8.14+rc1 failed at "./configure -configdir
        /tmp/coq-dune-issue/_opam/lib/coq/config -prefix /tmp/coq-dune-issue/_opam -mandir
        /tmp/coq-dune-issue/_opam/man -docdir /tmp/coq-dune-issue/_opam/doc -libdir
        /tmp/coq-dune-issue/_opam/lib/coq -datadir /tmp/coq-dune-issue/_opam/share/coq -coqide no
        -native-compiler no".

#=== ERROR while compiling coq.8.14+rc1 =======================================#
# context     2.1.0 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/coq/coq/archive/V8.14+rc1.tar.gz)
# path        /tmp/coq-dune-issue/_opam/.opam-switch/build/coq.8.14+rc1
# command     /tmp/coq-dune-issue/_opam/.opam-switch/build/coq.8.14+rc1/./configure -configdir /tmp/coq-dune-issue/_opam/lib/coq/config -prefix /tmp/coq-dune-issue/_opam -mandir /tmp/coq-dune-issue/_opam/man -docdir /tmp/coq-dune-issue/_opam/doc -libdir /tmp/coq-dune-issue/_opam/lib/coq -datadir /tmp/coq-dune-issue/_opam/share/coq -coqide no -native-compiler no
# exit-code   1
# env-file    /local/home/mackie/.opam/log/coq-22206-9f7e08.env
# output-file /local/home/mackie/.opam/log/coq-22206-9f7e08.out
### output ###
# Entering directory '/tmp/coq-dune-issue'
# Error: Program "./tools/configure/configure.exe" not found!



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build coq 8.14+rc1
└─ 
┌─ The following changes have been performed
│ ∗ install conf-findutils           1
│ ∗ install conf-gmp                 3
│ ∗ install dune                     2.9.1
│ ∗ install ocaml-secondary-compiler 4.08.1-1
│ ∗ install ocamlfind                1.9.1
│ ∗ install ocamlfind-secondary      1.9.1
│ ∗ install zarith                   1.12
└─ 
# Run eval $(opam env) to update the current shell environment
[NOTE] Pinning command successful, but your installed packages may be out of sync.

The former state can be restored with:
    /local/usr/local/bin/opam switch import
"/tmp/coq-dune-issue/_opam/.opam-switch/backup/state-20210922070820.export"
```

#### Coq Version

Tested with `8.14+rc1` and `8.14.dev`

With this edit, I get
```
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
⬇ retrieved coq.8.14+rc1  (cached)
[ERROR] The compilation of coq.8.14+rc1 failed at "make -j7".

#=== ERROR while compiling coq.8.14+rc1 =======================================#
# context     2.1.0 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/coq/coq/archive/V8.14+rc1.tar.gz)
# path        /tmp/coq-dune-issue/_opam/.opam-switch/build/coq.8.14+rc1
# command     /usr/bin/make -j7
# exit-code   2
# env-file    /local/home/mackie/.opam/log/coq-30303-ede2cb.env
# output-file /local/home/mackie/.opam/log/coq-30303-ede2cb.out
### output ###
# [...]
# make[1]: *** [Makefile.common:131: _build/install/default/bin/coqchk] Error 1
# Entering directory '/tmp/coq-dune-issue'
# Error: Don't know how to build _build/install/default/bin/coqqueryworker.opt
# make[1]: *** [Makefile.common:131: _build/install/default/bin/coqqueryworker.opt] Error 1
# Entering directory '/tmp/coq-dune-issue'
# Error: Don't know how to build _build/install/default/bin/coqtacticworker.opt
# make[1]: *** [Makefile.common:131: _build/install/default/bin/coqtacticworker.opt] Error 1
# Entering directory '/tmp/coq-dune-issue'
# Error: Don't know how to build _build/install/default/bin/coqproofworker.opt
# make[1]: *** [Makefile.common:131: _build/install/default/bin/coqproofworker.opt] Error 1
# make[1]: Leaving directory '/tmp/coq-dune-issue/_opam/.opam-switch/build/coq.8.14+rc1'
# make: *** [Makefile.make:122: submake] Error 2



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build coq 8.14+rc1
└─ 
```
This is the output of `opam show coq --raw`:
```
opam-version: "2.0"
name: "coq"
version: "8.14+rc1"
synopsis: "Formal proof management system"
description: """\
The Coq proof assistant provides a formal language to write
mathematical definitions, executable algorithms, and theorems, together
with an environment for semi-interactive development of machine-checked
proofs. Typical applications include the certification of properties of programming
languages (e.g., the CompCert compiler certification project and the
Bedrock verified low-level programming library), the formalization of
mathematics (e.g., the full formalization of the Feit-Thompson theorem
and homotopy type theory) and teaching."""
maintainer: "coqdev@inria.fr"
authors: "The Coq development team, INRIA, CNRS, and contributors."
license: "LGPL-2.1-only"
homepage: "https://coq.inria.fr/"
bug-reports: "https://github.com/coq/coq/issues"
depends: [
  "ocaml" {>= "4.05.0"}
  "ocamlfind" {build}
  "dune" {>= "2.5.1"}
  "conf-findutils" {build}
  "zarith" {>= "1.10"}
]
depopts: ["coq-native"]
build: [
  [
    "dune"
    "exec"
    "--root"
    "."
    "--"
    "./tools/configure/configure.exe"
    "-configdir"
    "%{lib}%/coq/config"
    "-prefix"
    prefix
    "-mandir"
    man
    "-docdir"
    doc
    "-libdir"
    "%{lib}%/coq"
    "-datadir"
    "%{share}%/coq"
    "-coqide"
    "no"
    "-native-compiler"
    "yes" {coq-native:installed}
    "no" {!coq-native:installed}
  ]
  [make "-j%{jobs}%"]
]
install: [make "install"]
dev-repo: "git+https://github.com/coq/coq.git"
url {
  src: "https://github.com/coq/coq/archive/V8.14+rc1.tar.gz"
  checksum:
    "sha512=53f193e28e8c9e5dffdb8bd33edbb2e38922983c85a60fc2800ba388a081e37be2add5cac566590c8de7ccd330e0f89d2b8749ef2e33302e2ae0cc595a8539e6"
}
```

--- a/Makefile.common
+++ b/Makefile.common
@@ -116,7 +116,7 @@ DOCGRAM_SOURCE_FILES=$(shell find $(addprefix $(COQ_SRC_DIR)/, doc/tools/docgram
 # Override for developer build [to get warn-as-error for example]
 _DDISPLAY?=quiet
 _DPROFILE?=$(CONFIGURE_DPROFILE)
-_DOPT:=--display $(_DDISPLAY) --profile=$(_DPROFILE)
+_DOPT:=--display=$(_DDISPLAY) $(_DPROFILE)
 _DBUILD:=$(FLOCK) .dune.lock dune build $(_DOPT)
 
 # We rerun dune when any of the source files have changed

--- a/configure
+++ b/configure
@@ -11,4 +11,4 @@ then
     exit 1
 fi
 
-dune exec -- $configure "$@"
+dune exec --root . -- $configure "$@"

--- a/tools/configure/cmdArgs.ml
+++ b/tools/configure/cmdArgs.ml
@@ -71,7 +71,7 @@ let default = {
     if os_type_win32 || os_type_cygwin then NativeNo else NativeOndemand;
   coqwebsite = "http://coq.inria.fr/";
   warn_error = false;
-  dune_profile = "release";
+  dune_profile = "--release";
   install_enabled = true;
 }
 
@@ -79,7 +79,7 @@ let devel state = { state with
   bin_annot = true;
   annot = true;
   warn_error = true;
-  dune_profile = "dev";
+  dune_profile = "--profile=dev";
   interactive = true;
   prefix = Some (Filename.concat (Sys.getcwd ()) "_build_vo/default");
   install_enabled = false;

